Program.cs

```diff
@@ -24,8 +31,46 @@ class MyWindow : Window {
       mStride = mBmp.BackBufferStride;
       image.Source = mBmp;
       Content = image;
+      MouseMove += OnMouseMove;
+      MouseLeftButtonDown += OnMouseLeftButtonDown;
+      //DrawMandelbrot (-0.5, 0, 1);
+   }
+   void OnMouseLeftButtonDown (object sender, MouseButtonEventArgs e) {
+      var pos = e.GetPosition (this);
+      if (mStartPt == null)
+         mStartPt = new (pos.X, pos.Y);
+      else {
+         mEndPt = new (pos.X, pos.Y);
+         DrawLine (mStartPt, mEndPt);
+         mStartPt = null;
+      }
+   }
 
-      DrawMandelbrot (-0.5, 0, 1);
+   void DrawLine (Point2D startPt, Point2D endPt) {
+      var pts = GetCLPoints (startPt, endPt);
+      foreach (var pt in pts) {
+         try {
+            mBmp.Lock ();
+            mBase = mBmp.BackBuffer;
+            SetPixel (pt.X, pt.Y, 255);
+            mBmp.AddDirtyRect (new Int32Rect (pt.X, pt.Y, 1, 1));
+         } finally {
+            mBmp.Unlock ();
+         }
+      }
+   }
+
+   /// <summary> Returns list of co-linear points lie between start point and end point </summary>
+   List<Point2D> GetCLPoints (Point2D startPt, Point2D endPt) {
+      List<Point2D> pts = new ();
+      var dist = Point2D.Distance (startPt, endPt);
+      var dx = (endPt.X - startPt.X) / dist;
+      var dy = (endPt.Y - startPt.Y) / dist;
+      for (int i = 1; i <= (int)dist; i++) {
+         var pt = new Point2D (startPt.X + dx * i, startPt.Y + dy * i);
+         pts.Add (pt);
+      }
+      return pts;
    }

```

Comment:
The lines when rendered don't look smooth. They look very jagged. Can you please try to render more smoothened lines?
<img width="317" alt="image" src="https://github.com/arivazhagan-tmm/Graphics23/assets/75829870/81588f0a-c9cf-46f2-8e5f-5fb5b87e1656">

Reply:
Yeah sure.. will try and update it..